code/modules/library/library_catalog.dm

```diff
@@ -0,0 +1,616 @@
+///Maximum number of books that can be uploaded by a single ckey
+#define MAX_PLAYER_UPLOADS 5
+
+/*
+ * #Library Catalog
+ *
+ * This datum forms the basis for the entire library system, one is created at roundstart and stored to a global variable
+ * It holds lists for all predefined report and book categories, books flagged during the round, and all programmatic
+ * books.
+ *
+ * Additionally, ALL library SQL queries are handled in this datum. This is intentional so that we do not have multiple SQL
+ * queries trying to do the same thing but in 4-5 different dm files. This datum is split into a specific
+ * structure: 1) Defining Library Lists 2) Get Procs that get information from the catalog or DB 3) Send/Update procs that
+ * perform changes to the Database. Don't mix their functionalities together.
+ */
+/datum/library_catalog
+	///Lists of all reported books in the current round
+	var/list/flagged_books = list()
+	///List of all programmatic books, automatically generated upon New()
+	var/list/books = list()
+	///List of all report categories, automatically generated upon New()
+	var/list/report_types = list()
+	///List of all book categories, automatically generated upon New()
+	var/list/categories = list()
+
+/datum/library_catalog/New()
+
+	//Building a list of all the reasons that players can report books, used for report menu + logging reports in DB
+	//Cat ID needs to be unique to category, description can be changed here without issues anywhere else
+	report_types = list(
+		DEFINE_CATEGORY(LIB_REPORT_HATESPEECH, "Hatespeech or Slur Usage"),
+		DEFINE_CATEGORY(LIB_REPORT_EROTICA, "Erotica or Sexual Content"),
+		DEFINE_CATEGORY(LIB_REPORT_OOC, "Out of Character Information"),
+		DEFINE_CATEGORY(LIB_REPORT_COPYPASTA, "Copypastas or Spam"),
+		DEFINE_CATEGORY(LIB_REPORT_BLANK , "Blank or No Content"),
+		DEFINE_CATEGORY(LIB_REPORT_NOEFFORT , "Very Low Effort Content"),
+		DEFINE_CATEGORY(LIB_REPORT_OTHER  , "Other Reason not Specified"), //required
+	)
+
+	//building a list of all categories, used for searching, Cat ID needs to be unique to category
+	categories = list(
+		DEFINE_CATEGORY(LIB_CATEGORY_FICTION, "Fiction"),
+		DEFINE_CATEGORY(LIB_CATEGORY_NONFICTION, "Non-Fiction"),
+		DEFINE_CATEGORY(LIB_CATEGORY_RELIGION, "Religious"),
+		DEFINE_CATEGORY(LIB_CATEGORY_FANTASY, "Fantasy"),
+		DEFINE_CATEGORY(LIB_CATEGORY_HORROR, "Horror"),
+		DEFINE_CATEGORY(LIB_CATEGORY_ROMANCE, "Romance"),
+		DEFINE_CATEGORY(LIB_CATEGORY_MYSTERY, "Mystery"),
+		DEFINE_CATEGORY(LIB_CATEGORY_ADVENTURE, "Adventure"),
+		DEFINE_CATEGORY(LIB_CATEGORY_HISTORY, "History"),
+		DEFINE_CATEGORY(LIB_CATEGORY_PHILOSOPHY, "Philosophy"),
+		DEFINE_CATEGORY(LIB_CATEGORY_DRAMA, "Drama and Thriller"),
+    	DEFINE_CATEGORY(LIB_CATEGORY_EXPERIMENT, "Experiment Notes"),
+		DEFINE_CATEGORY(LIB_CATEGORY_LEGAL, "Legal Document"),
+		DEFINE_CATEGORY(LIB_CATEGORY_BIOGRAPHY, "Biography"),
+		DEFINE_CATEGORY(LIB_CATEGORY_GUIDE, "Guides and References"),
```

Comment:
Spacing here? Or is GitHub just rendering it badly.

Reply:
this is github rendering badly. There should be empty lines at L55 and L66